internal/polkavm/interpreter/mutator.go

```diff
@@ -498,37 +498,37 @@ func (m *Mutator) CmovIfNotZero(d polkavm.Reg, s, c polkavm.Reg) {
 	}
 	m.instance.NextOffsets()
 }
-func (m *Mutator) RotL64(d polkavm.Reg, s, c polkavm.Reg) {
-	panic("todo: implement")
+func (m *Mutator) RotL64(d polkavm.Reg, s1, s2 polkavm.Reg) {
+	m.set64(d, bits.RotateLeft64(m.get64(s1), int(m.get64(s2))))
 }
-func (m *Mutator) RotL32(d polkavm.Reg, s, c polkavm.Reg) {
-	panic("todo: implement")
+func (m *Mutator) RotL32(d polkavm.Reg, s1, s2 polkavm.Reg) {
+	m.set32(d, bits.RotateLeft32(m.get32(s1), int(m.get32(s2))))
```

Comment:
Shouldn't the rotation amount come from `s1`?
```
m.set32(d, bits.RotateLeft32(m.get32(s1), -int(m.get32(s1))))

Reply:
Confirmed in matrix, it's a typo in the GP and will be fixed in later versions